[Bug]: Support riscv64, ppc64el, s390x etc -- missing `getSyscallArgs`

Building in-toto/go-witness 0.9.2 in Debian on fails like this for non-amd64/arm64/i386 platforms:

```
1269s github.com/in-toto/go-witness/attestation/commandrun
1272s # github.com/in-toto/go-witness/attestation/commandrun
1272s src/github.com/in-toto/go-witness/attestation/commandrun/tracing_linux.go:167:14: undefined: getSyscallArgs
1272s src/github.com/in-toto/go-witness/attestation/commandrun/tracing_linux.go:168:15: undefined: getSyscallId
1272s src/github.com/in-toto/go-witness/attestation/commandrun/tracing_linux.go:280:9: undefined: getNativeUint
```

The cause seems obvious looking at:

https://github.com/in-toto/go-witness/tree/main/attestation/commandrun

It has implementations for amd64, arm, arm64, and i386 -- but not any other platform.  And the `tracing_unsupported.go` file does not implement these APIs and is for non-linux platforms only.

Could you implement stubs that works everywhere?  Or at least on all linux.

Thanks,
Simon

--- a/attestation/commandrun/tracing_linux.go
+++ b/attestation/commandrun/tracing_linux.go
@@ -441,8 +441,8 @@ func (ctx *ptraceContext) readSyscallReg(pid int, addr uintptr, n int) (string, 
 	data := make([]byte, n)
 	localIov := unix.Iovec{
 		Base: &data[0],
-		Len:  getNativeUint(n),
 	}
+	localIov.SetLen(n)
 
 	removeIov := unix.RemoteIovec{
 		Base: addr,

--- a/attestation/commandrun/tracing_linux_386.go
+++ b/attestation/commandrun/tracing_linux_386.go
@@ -34,7 +34,3 @@ func getSyscallArgs(regs unix.PtraceRegs) []uintptr {
 		uintptr(regs.Ebp),
 	}
 }
-
-func getNativeUint(n int) uint32 {
-	return uint32(n)
-}

--- a/attestation/commandrun/tracing_linux_amd64.go
+++ b/attestation/commandrun/tracing_linux_amd64.go
@@ -34,7 +34,3 @@ func getSyscallArgs(regs unix.PtraceRegs) []uintptr {
 		uintptr(regs.R9),
 	}
 }
-
-func getNativeUint(n int) uint64 {
-	return uint64(n)
-}

--- a/attestation/commandrun/tracing_linux_arm.go
+++ b/attestation/commandrun/tracing_linux_arm.go
@@ -35,7 +35,3 @@ func getSyscallArgs(regs unix.PtraceRegs) []uintptr {
 		uintptr(regs.Uregs[5]),
 	}
 }
-
-func getNativeUint(n int) uint32 {
-	return uint32(n)
-}

--- a/attestation/commandrun/tracing_linux_arm64.go
+++ b/attestation/commandrun/tracing_linux_arm64.go
@@ -34,7 +34,3 @@ func getSyscallArgs(regs unix.PtraceRegs) []uintptr {
 		uintptr(regs.Regs[5]),
 	}
 }
-
-func getNativeUint(n int) uint64 {
-	return uint64(n)
-}

--- a/attestation/commandrun/tracing_linux_unsupported_arch.go
+++ b/attestation/commandrun/tracing_linux_unsupported_arch.go
@@ -1,0 +1,31 @@
+// Copyright 2021 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux && !amd64 && !arm64 && !arm && !386
+
+package commandrun
+
+import (
+	"golang.org/x/sys/unix"
+)
+
+// Fallback syscall register decoding for Linux architectures that do not yet
+// have architecture-specific ptrace register mappings.
+func getSyscallId(regs unix.PtraceRegs) int {
+	return -1
+}
+
+func getSyscallArgs(regs unix.PtraceRegs) []uintptr {
+	return nil
+}

